pkg/event/controller/controller.go

```diff
@@ -43,15 +44,36 @@ type specification struct {
 	Namespace string
 }
 
+type handlerStateType struct {
```

Comment:
We could just name it `handlerState` but it's nice to distinguish it from variables of the same name.  I've seen the use of `Type` in K8s code so I borrowed it (I've used it before as well) - it is actually describing the type. `Impl` seems Java-ish but I can change it - I'm fine either way (or name it `handlerState`).

Reply:
Right, I didn’t bother mentioning `handlerState` because of the variable name conflict! I prefer `Impl` personally, especially as more type parameters start appearing in various places. The naming in Kubernetes seems to be shifting too, at least in discussions I’ve been having recently people talk about `Impl`, and it’s about 50/50 now.